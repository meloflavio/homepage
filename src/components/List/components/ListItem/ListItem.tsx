import { LiHTMLAttributes } from 'react';

import * as S from './styles';

type ListItemProps = LiHTMLAttributes<HTMLLIElement> & {
  children: React.ReactNode;
};

export const ListItem = ({ children, ...rest }: ListItemProps) => {
  return <S.Container {...rest}>{children}</S.Container>;
};
