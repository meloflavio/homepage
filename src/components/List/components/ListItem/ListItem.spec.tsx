import { render, screen } from '@testing-library/react';

import { ListItem } from './ListItem';

describe('<ListItem />', () => {
  it('should render a listItem', () => {
    const children = 'children';

    const { container } = render(<ListItem>{children}</ListItem>);

    const childrenElement = screen.getByText(children);

    expect(childrenElement).toBeVisible();
    expect(container).toMatchSnapshot();
  });
});
