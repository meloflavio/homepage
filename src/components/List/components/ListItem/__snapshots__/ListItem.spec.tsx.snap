// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ListItem /> should render a listItem 1`] = `
<div>
  <li
    class="text-md relative px-2 font-light transition-colors duration-200 hover:text-link"
  >
    children
  </li>
</div>
`;
